client/internal/utils/httpclient.go

```diff
@@ -0,0 +1,13 @@
+package utils
+
+import "net/http"
+
+func NewHttpClient() *http.Client {
+	client := &http.Client{}
+	// Clone the default transport to avoid being affected by global state changes.
+	if tr, ok := http.DefaultTransport.(*http.Transport); ok {
```

Comment:
Are we doing this specifically so that `http.DefaultTransport` is not nil? or are we able to just use `&http.Client{}` instead of a function to create a client?